### ISSUE TITLE
Repeating Table - Not reordering properly

### ISSUE THREAD

**Issue by grahamsinger** [tsgrp/ActiveWizard#252]
Reordering rows in a repeating question table doesn't seem to work as expected in some situations.  (Not sure if this is related to the problems we are having in #251.)

See here as moving the first row to the end caused the last row to jump to the first position, leaving the second row in its same position.
![repeatingtablebroken](https://cloud.githubusercontent.com/assets/8784597/9178172/67d7b3c2-3f5b-11e5-93b1-46027305bbae.gif)

Based on how the rows shift when dragging, it looks like the first row would move to the bottom and the other two would just shift up.  However, in the code it looks like it is intended to simply swap, as the .gif shows above: 

```
swapQuestions: function(modelA, modelB){
        var indexA = this.indexOf(modelA);
        var indexB = this.indexOf(modelB);
        this.models[indexA] = modelB;
        this.models[indexB] = modelA;
}
```

@gsteimer @mikeblum Just curious as to the expected functionality.

**Comment by grahamsinger** [comment]
To update, this is an issue.  It should not use the "swapping" logic to reorder. It should function like any other sortable list we use (i.e. in the above example, the 2nd row would shift up to the 1st instead of staying in its original position).

**Comment by grahamsinger** [comment]
Fixed with revision 12834

We were just swapping the rows before when we rearranged instead of having the rows adjust as expected and indicated by jQuery sortable.  Also, we were never reordering the `questionGroups[]` so when we went to delete after rearranging the rows, it would be deleting the wrong row. So now, we ensure that both the individual question models as well as the question groups are in the right order so both rearranging, deleting.  Lastly, previously we were just getting the last element in the `questionGroups[]` assuming that it would have the highest row number `..._rX`.  As stated before, this wasn't as issue because we never actually reorganized that array to match what was going on in the table.  I changed the `getLastQuestionGroup()` to make sure to find the highest row number because adding rows would fail otherwise.

Unit tested.

CR: Mike B
